src/components/symptom-shark/container/ReportBuilder/ReportBuilder.tsx

```diff
@@ -0,0 +1,221 @@
+    const [selectedSymptoms, setSelectedSymptoms] = useState<string[]>([]);
+    const [selectedTreatments, setSelectedTreatments] = useState<string[]>([]);
+    const [includeDailyOverallExperience, setIncludeDailyOverallExperience] = useState<boolean>(true);
+    const [includeNotes, setIncludeNotes] = useState<boolean>(false);
+    const [buildingReport, setBuildingReport] = useState<boolean>(false);
+    const report = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        initialize();
+        MyDataHelps.on("applicationDidBecomeVisible", initialize);
+        return () => {
+            MyDataHelps.off("applicationDidBecomeVisible", initialize);
+        }
+    }, []);
+
+    function initialize() {
+        if (props.previewState == "default") {
+            setConfiguration({
+                participantID: "1",
+                symptoms: demoSymptoms,
+                treatments: demoTreatments
+            });
+            setLogEntries(demoLogEntries);
+            return;
+        }
+
+        symptomSharkData.getConfiguration().then(function (configuration) {
+            symptomSharkData.getDailyLogEntries().then(function (logEntries) {
+                setConfiguration(configuration);
+                setLogEntries(logEntries);
+            });
+        });
+    }
+
+    if (!configuration || !logEntries) return <div ref={props.innerRef}><LoadingIndicator /></div>;
+
+    function toggleSymptom(item: SymptomConfiguration) {
+        var newSelectedItems = [...selectedSymptoms];
+        if (newSelectedItems.indexOf(item.id) != -1) {
+            newSelectedItems.splice(newSelectedItems.indexOf(item.id), 1);
+        } else {
+            newSelectedItems.push(item.id);
+        }
+        setSelectedSymptoms(newSelectedItems);
+    }
+
+    function toggleTreatment(item: TreatmentConfiguration) {
+        var newSelectedItems = [...selectedTreatments];
+        if (newSelectedItems.indexOf(item.id) != -1) {
+            newSelectedItems.splice(newSelectedItems.indexOf(item.id), 1);
+        } else {
+            newSelectedItems.push(item.id);
+        }
+        setSelectedTreatments(newSelectedItems);
+    }
+
+    function buildReport() {
+        setBuildingReport(true);
+
+        //hack: get the styles from the document and add them to the report so they are included in the PDF
+        var documentStyles = document.head.getElementsByTagName("style");
+        var html = "";
+        for (var i = 0; i < documentStyles.length; i++) {
+            html += documentStyles[i].outerHTML;
+        }
+        html += report.current!.innerHTML;
+
+        renderPdf(html, configuration!.participantID).then(function () {
+            setBuildingReport(false);
+        });
+    }
+
+    function getSelectedSymptomConfigurations() {
+        if (!includeSymptoms) { return []; }
+        return configuration!.symptoms.filter((s) => !selectedSymptoms.length || selectedSymptoms.indexOf(s.id) != -1);
```

Comment:
Probably doesn't make much difference performance-wise, but I think you can use `selectedSymptoms.includes(s.id)` instead of `indexOf` for these checks.